visual-studio-vstsbuildagent-vm/InstallVSTSAgent.ps1

```diff
@@ -81,7 +83,88 @@ Write-Verbose "Configuring agent" -Verbose
 # Set the current directory to the agent dedicated one previously created.
 Push-Location -Path $agentInstallationPath
 
-.\config.cmd --unattended --url $serverUrl --auth PAT --token $PersonalAccessToken --pool $PoolName --agent $AgentName --runasservice
+if ($runAsAutoLogon -ieq "true")
+{
+  # Create a PS session for the user to trigger the creation of the registry entries required for autologon
+  $computerName = "localhost"
+  $password�=�ConvertTo-SecureString�$vmAdminPassword -AsPlainText�-Force
+  if ($vmAdminUserName.Split("\").Count -eq 2)
+  {
+    $domain = $vmAdminUserName.Split("\")[0]
+    $userName = $vmAdminUserName.Split('\')[1]
+  }
+  else
+  {
+    $domain = Hostname
+    $userName = $vmAdminUserName
+  }
+  $credentials =�New-Object�System.Management.Automation.PSCredential("$domain\\$userName", $password)
+  Enter-PSSession�-ComputerName�$computerName�-Credential�$credentials
```

Comment:
>Enter-PSSession [](start = 2, length = 15)

is there a possibility of a failure here..?

Reply:
There is, but if it fails here it should just abort since as of now the user hive wouldn't be created. Not very sure if i should let it proceed or not